src/js/scripts.js

```diff
@@ -80,164 +80,162 @@ import msToMinutesAndSeconds from '../modules/modal-timeout/utils/msToMinutesAnd
         }
 
         if (modalElements.length) {
+            sessionData = getSessionData();
+            const sessionAlive = sessionData && sessionData.alive;
+
+            // haven't hit `/apply` yet, or the application is completed
+            // there is no session expiry cookie
+            if (!sessionAlive) {
+                return;
+            }
+
             window.document.addEventListener('visibilitychange', () => {
                 documentVisible = window.document.visibilityState === 'visible';
                 if (documentVisible) {
-                    sessionData = getSessionData();
-                    checkShouldOpenModal();
+                    if (sessionAlive) {
```

Comment:
remove this check

Reply:
Refactored the code and made new improvements which we discussed yesterday.